src/Tools/dotnet-watch/BrowserRefresh/src/WebSocketScriptInjection.cs

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Watch.BrowserRefresh
+{
+    /// <summary>
+    /// Helper class that handles the HTML injection into
+    /// a string or byte array.
+    /// </summary>
+    public static class WebSocketScriptInjection
+    {
+        private const string BodyMarker = "</body>";
+
+        private static readonly byte[] _bodyBytes = Encoding.UTF8.GetBytes(BodyMarker);
+        private static readonly byte[] _scriptInjectionBytes = Encoding.UTF8.GetBytes(GetWebSocketClientJavaScript());
+
+        public static async ValueTask<bool> TryInjectLiveReloadScriptAsync(byte[] buffer, int offset, int count, Stream baseStream)
+        {
+            var index = buffer.AsSpan(offset, count).LastIndexOf(_bodyBytes);
```

Comment:
Yup. It's entirely possible to see writes like this. Fixing it is much more work. I'd like to see what feedback we get. Given that the other middleware has existed for a while, and they didn't think it was worthwhile address it, this might be ok.